app/src/main/java/com/nervousfish/nervousfish/activities/RhythmCreateActivity.java

```diff
@@ -0,0 +1,186 @@
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_rhythm_create);
+        final Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar_create_rhythm);
+        setSupportActionBar(toolbar);
+
+        if (getSupportActionBar() != null) {
+            getSupportActionBar().setDisplayShowTitleEnabled(false);
+        }
+
+        final Intent intent = this.getIntent();
+        this.serviceLocator = (IServiceLocator) intent.getSerializableExtra(ConstantKeywords.SERVICE_LOCATOR);
+
+        startButton = (Button) this.findViewById(R.id.start_recording_button);
+        stopButton = (Button) this.findViewById(R.id.stop_recording_button);
+        doneButton = (Button) this.findViewById(R.id.done_tapping_button);
+
+        LOGGER.info("RhythmActivity started");
+    }
+
+    /**
+     * Gets triggered when the tap screen is clicked.
+     *
+     * @param v - the {@link View} clicked
+     */
+    public void onTapClick(final View v) {
+        LOGGER.info("Tapped");
+        if (tapCombination != null && startButton.getVisibility() == View.GONE) {
+            tapCombination.add(new SingleTap(new Timestamp(System.currentTimeMillis())));
+        }
+    }
+
+    /**
+     * Gets triggered when the done button is clicked.
+     *
+     * @param v - the {@link View} clicked
+     */
+    public void onDoneCreatingRhythmClick(final View v) {
+        LOGGER.info("Done tapping button clicked");
+        try {
+            final Profile myProfile = this.serviceLocator.getDatabase().getProfiles().get(0);
+
+            LOGGER.info("Sending my profile with name: " + myProfile.getName() + ", public key: "
+                    + myProfile.getPublicKey().toString());
+            final Contact myProfileAsContact = new Contact(myProfile.getName(), new SimpleKey("simplekey", "73890ien"));
+            this.serviceLocator.getBluetoothHandler().send(myProfileAsContact);
+        } catch (IOException e) {
+            LOGGER.error("Could not send my contact to other device " + e.getMessage());
+        }
+        final Intent intent = new Intent(this, WaitActivity.class);
+        intent.putExtra(ConstantKeywords.SERVICE_LOCATOR, serviceLocator);
+        intent.putExtra(ConstantKeywords.WAIT_MESSAGE, this.getString(R.string.wait_message_partner_rhythm_tapping));
+        intent.putExtra(ConstantKeywords.DATA_RECEIVED, dataReceived);
+        intent.putExtra(ConstantKeywords.TAP_DATA, (ArrayList) tapCombination);
+        this.startActivityForResult(intent, ConstantKeywords.START_RHYTHM_REQUEST_CODE);
+    }
+
+    /**
+     * Gets triggered when the start recording button is clicked.
+     *
+     * @param v - the {@link View} clicked
+     */
+    public void onStartRecordingClick(final View v) {
+        LOGGER.info("Start Recording clicked");
+        tapCombination = new ArrayList<>();
+
+        startButton.setVisibility(View.GONE);
+
+        stopButton.setVisibility(View.VISIBLE);
+
```

Comment:
I don't think it adds any readability since it concerns single lines. If you want to make clear these are discreet steps in the program, consider adding comments to make the distinction clear